src/encoded/static/components/antibody.js

```diff
@@ -67,18 +67,28 @@ const LotComponent = (props, reactContext) => {
     const geneTips = [];
     targetKeys.forEach((key, i) => {
         const scientificName = targets[key].organism.scientific_name;
-        const geneName = targets[key].gene_name;
+        const geneName = targets[key].genes.map(
+            gene => gene.symbol
+        );
 
         // Add to the information on organisms from the targets
         organismComponents.push(<span key={key}>{i > 0 ? <span> + <i>{scientificName}</i></span> : <i>{scientificName}</i>}</span>);
         organismTerms.push(`targets.organism.scientific_name=${scientificName}`);
         organismTips.push(scientificName);
 
         // Add to the information on gene names from the targets
-        if (geneName && geneName !== 'unknown') {
-            geneComponents.push(<span key={key}>{i > 0 ? <span> + {geneName}</span> : <span>{geneName}</span>}</span>);
-            geneTerms.push(`targets.gene_name=${geneName}`);
-            geneTips.push(geneName);
+        if (geneName.length) {
+            geneName.map(
```

Comment:
.map() is really designed to convert one array to another, not as a generic looping structure. You can replace with .forEach() so the fact that no conversion happens is clear to others (though you’ll need to convert to an explicit return).

_But_ this also looks like it _might_ be an actual mapping operation. It _looks_ like you can keep .map() but rather than pushing geneComponents you can just return the span, and assign the result of .map() to geneComponents.

You’re probably also getting a Javascript warning in the console, if not an error, because you’re assigning the same `key` to multiple `<spans>`. If `symbol` is unique in each iteration, you can use `key={symbol}` instead to get rid of the error.

Reply:
Thank you very much for all these comments, especially the detailed explanations. They are very helpful and let me learn a lot!

Reply:
Pushed a new commit. Could you help me check if the way of defining and assigning variables is good practice?